NebulaHost/PacketProcessors/Universe/DysonSphereRequestProcessor.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.IO;
+using System.IO.Compression;
+using LZ4;
+using NebulaModel.Attributes;
+using NebulaModel.Networking;
+using NebulaModel.Packets.Universe;
+using NebulaModel.Packets.Processors;
+using NebulaModel.Logger;
+
+namespace NebulaHost.PacketProcessors.Universe
+{
+    [RegisterPacketProcessor]
+    public class DysonSphereRequestProcessor : IPacketProcessor<DysonSphereLoadRequest>
+    {
+        public void ProcessPacket(DysonSphereLoadRequest packet, NebulaConnection conn)
+        {
+            DysonSphere dysonSphere = GameMain.data.CreateDysonSphere(packet.StarIndex);
+
+            using (MemoryStream ms = new MemoryStream())
+            {
+                using (LZ4Stream ls = new LZ4Stream(ms, CompressionMode.Compress))
+                using (BufferedStream bs = new BufferedStream(ls, 8192))
+                using (BinaryWriter bw = new BinaryWriter(bs))
+                {
+                    dysonSphere.Export(bw);
+                }
+                conn.SendPacket(new DysonSphereData(packet.StarIndex, ms.ToArray()));
```

Comment:
Seems weird to have to convert this to an array when we just want to transmit data over a stream. Can't we use CopyTo after resetting the position to 0

Reply:
Actually, I was just following the same approach that is already in [FactoryLoadRequestProcessor](https://github.com/hubastard/nebula/blob/0f1ba59d099e9d3b908fcf9c78c25ed982aed3bd/NebulaHost/PacketProcessors/Planet/FactoryLoadRequestProcessor.cs#L28) and [PlanetDataRequestProcessor](https://github.com/hubastard/nebula/blob/0f1ba59d099e9d3b908fcf9c78c25ed982aed3bd/NebulaHost/PacketProcessors/Planet/PlanetDataRequestProcessor.cs#L59).

Reply:
Understood, I can take a look at this after, no worries.

Reply:
@Therzok Were you implying that you will look at this after the merge, or you wanted to take a look before I merge this?

Reply:
After